Assets/Scripts/Fire.cs

```diff
@@ -6,38 +6,54 @@ public class Fire : MonoBehaviour {
     public float speedDecreasing;
     public float minScale;
     public int densityFire;
+    public float damageFire;
+    public GameObject col;
     private ParticleSystem part;
     private ParticleSystem.ShapeModule sh;
     private ParticleSystem.EmissionModule em;
 
     void Start() {
+        col.transform.localScale = Vector3.one * scaleFire * 100;
         part = GetComponent<ParticleSystem>();
         sh = part.shape;
         em = part.emission;
-        sh.scale = Vector3.ClampMagnitude(Vector3.one, scaleFire);
+        sh.scale = Vector3.one * scaleFire;
         em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(scaleFire, 3) * densityFire);
     }
 
     void Update() {
         if (!part.isStopped) {
-            var collids = Physics.OverlapSphere(transform.position, sh.scale.magnitude);
-            collids = collids.Where(x => x.CompareTag("Item")).ToArray<Collider>();
-
-            foreach (var collid in collids) {
-                var item = collid.gameObject.GetComponent<ItemAssociation>().item;
-                if (item.fuelSize > 0) {
-                    sh.scale += Vector3.ClampMagnitude(Vector3.one, item.fuelSize);
-                    em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3));
-                    Destroy(collid.gameObject);
-                }
-            }
-
             if (sh.scale.magnitude < minScale) {
                 part.Stop();
             } else if (!part.isStopped) {
-                sh.scale -= Vector3.ClampMagnitude(Vector3.one, speedDecreasing * Time.deltaTime);
+                col.transform.localScale -= Vector3.one * speedDecreasing * 100 * Time.deltaTime;
+
+                sh.scale -= Vector3.one * speedDecreasing * Time.deltaTime;
                 em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3) * densityFire);
             }
         }
     }
+
+    void OnTriggerEnter(Collider collider) {
+        Debug.Log("Something is in the fire");
+        if (collider.gameObject.CompareTag("Item")) {
+            var item = collider.gameObject.GetComponent<ItemAssociation>().item;
+            if (item.fuelSize > 0) {
+                col.transform.localScale += Vector3.one * item.fuelSize * 100;
+
+                sh.scale += Vector3.one * item.fuelSize;
+                em.rateOverTime = (ParticleSystem.MinMaxCurve)(System.Math.Pow(sh.scale.magnitude, 3));
+                Destroy(collider.gameObject);
+            }
+
+        } else if (collider.gameObject.CompareTag("Player")) {
+            collider.gameObject.GetComponent<PlayerMovement>().TakeDamage(damageFire * Time.deltaTime);
+        }
+    }
+
+    private void OnTriggerStay(Collider other) {
```

Comment:
This trigger method is a lot better than the previous sphere overlap, awesome :) One minor bug, you keep taking damage once the fire went out (maybe make damage relative to the size/distance of the fire? if feasible)

Also if you can leave a //todo here to remind me that I need to add damage to animals as well once we merge